vscode-lean4/src/utils/leanpkg.ts

```diff
@@ -108,22 +109,28 @@ export class LeanpkgService implements Disposable {
         }
     }
 
-    private findLakeFile(packageUri: Uri) : Uri | null {
+    private async findLakeFile(packageUri: Uri) : Promise<Uri | null> {
         const fullPath = Uri.joinPath(packageUri, this.lakeFileName);
         const url = fullPath.fsPath;
-        if (fs.existsSync(url)) {
+        if(await fsExistHelper(url)) {
             return fullPath;
         }
         return null;
     }
 
     // Return file contents with whitespace normalized.
-    private readWhitespaceNormalized(fileUri: Uri) : string {
-        const contents = fs.readFileSync(fileUri.fsPath).toString();
-        // ignore whitespace changes by normalizing whitespace.
-        const re = /[ \t\r\n]+/g
-        const result = contents.replace(re, ' ');
-        return result.trim();
+    private async readWhitespaceNormalized(fileUri: Uri) : Promise<string> {
+        try{
+            const contents = (await workspace.fs.readFile(fileUri)).toString();
+            // ignore whitespace changes by normalizing whitespace.
+            const re = /[ \t\r\n]+/g
+            const result = contents.replace(re, ' ');
+            return result.trim();
+        }
+        catch(ex) {
+            // In case there is an error in the read
+            throw ex;
```

Comment:
There is no point to this try/catch block, it has the same semantics as if you had no try catch, but perhaps the right semantic is to NOT throw and return empty string, then handleLakeFileChanged can continue as normal and not have to worry about exceptions.